storage/backend/ceph/ceph.go

```diff
@@ -41,6 +42,21 @@ type Driver struct {
 	mountpath string
 }
 
+func (c *Driver) mkMountPath(poolName, intName string) (string, error) {
+	// Directory to mount the volume
+	volumePath := filepath.Join(c.mountpath, mkpool(poolName, intName))
+	rel, err := filepath.Rel(c.mountpath, volumePath)
+	if err != nil || strings.Contains(rel, "..") {
+		return "", errors.MountFailed.Combine(errored.Errorf("Calculated volume path would escape subdir jail: %v", volumePath))
```

Comment:
At least for bind mounts, symlinks are resolved during mounts:

```sh
root:/tmp$ mkdir realdir
root:/tmp$ touch realdir/data.txt
root:/tmp$ ln -s realdir symlink
root:/tmp$ mkdir mountpoint
root:/tmp$ mount --bind /tmp/symlink /tmp/mountpoint/
root:/tmp$ ls -l /tmp/mountpoint/
data.txt
```

I have no idea if it applies at all to what we're doing, but I was curious.